src/components/VDataTable/mixins/body.js

```diff
@@ -28,7 +33,7 @@ export default {
         on: ExpandTransitionGenerator('datatable__expand-col--expanded')
       }, children)
 
-      return this.genTR([transition], { class: 'datatable__expand-row' })
+      return this.nowrapExpanded ? children : this.genTR([transition], { class: 'datatable__expand-row' })
```

Comment:
Move the return if nowrapExpanded to above line 27 so we're not creating an unused transition-group every time.